source/loader/lib_loader.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#include <array>
+
+#include "ur_api.h"
+#include "ur_ddi.h"
+#include "ur_object.hpp"
+
+#if defined(_WIN32)
```

Comment:
as a general rule, we should be using platform ifdefs very sparingly. In this case, I suggest creating a simple abstraction for managing dynamic libraries and then providing platform-specific implementations.

Reply:
Created files for linux- and windows-specific functions.